File: packages/javascript/lib/array/arrShuffle.ts
export function arrShuffle(array: any[]): any[] {
	// Shuffle a copy. Returning the input meant the caller's array was reordered in place.
	const newArray = [...array];

	for (let i = newArray.length - 1; i > 0; i -= 1) {
		const j = Math.floor(Math.random() * (i + 1));

		[newArray[i], newArray[j]] = [newArray[j], newArray[i]];
	}

	return newArray;
}

File: packages/python/qsu/array/arrShuffle.py
import random


def arrShuffle(array: list) -> list:
	# Shuffle a copy and always return a list. Returning `array[0]` for a single element
	# broke the return type, and assigning `newArray = array` reordered the caller's list.
	newArray = list(array)

	random.shuffle(newArray)

	return newArray
